File: js/카드 놓기(5568) - 재귀.js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const n = Number(input[0]);
const k = Number(input[1]);
const arr = input.slice(2, 2+n).map(Number);

let visited = Array.from({length: n}, () => false);

let card_list = [];
let result = [];

const backtracking = () => {
    if (card_list.length === k) {
        if (!result.includes(card_list.join(''))) {
            result.push(card_list.join(''));
        }
        return;
    }
    for (let i=0; i<arr.length; i++) {
        if (!visited[i]) {
            visited[i] = true;
            card_list.push(arr[i]);
            backtracking();
            visited[i] = false;
            card_list.pop();
        }
    }
}

backtracking();
console.log(result.length);

File: python/카드 놓기(5568) - 재귀.py
import sys
input = sys.stdin.readline

# 재귀활용법
# 카드를 뽑고 리스트에 저장
# 개수 + 1
# 만약 

n = int(input())
k = int(input())
arr = []
for _ in range(n):
    arr.append(int(input()))

card_list = []
visited = [False for _ in range(n)]

result = []

def backtracking():
    if len(card_list) == k:
        if ''.join(map(str, card_list)) not in result:
            result.append(''.join(map(str, card_list)))
        return
    
    for i in range(len(arr)):
        if not visited[i]:
            # 방문 처리
            visited[i] = True
            card_list.append(arr[i])
            backtracking()
            card_list.pop()
            visited[i] = False

backtracking()

print(len(result))